File: Updated/intro-data-structures/solutions/solution-set.js
var Set = function(){
  var set = Object.create(setProps);
  set._members = undefined;
  /* START SOLUTION */
  set._members = {};
  /* END SOLUTION */
  return set;
};

var setProps = {};

setProps.add = function(item){
  /* START SOLUTION */
  this._members[item] = true;
  /* END SOLUTION */
};

setProps.contains = function(item){
  /* START SOLUTION */
  return !!this._members[item];
  /* END SOLUTION */
};

setProps.remove = function(item){
  /* START SOLUTION */
  delete this._members[item];
  /* END SOLUTION */
};

File: Updated/intro-data-structures/solutions/solution_set.py
class Set(object):
	def __init__(self):
		self.set = {}

	def add(self, item):
		# SOLUTION START
		self.set[item] = True
		# SOLUTION END

	def contains(self, item):
		# SOLUTION START
		if len(self.set) == 0:
			print 'Add items to the set!'

		# Built in keyerror if key does not exist
		if self.set[item]:
			return self.set[item]
		# SOLUTION END

	def remove(self, item):
		# SOLUTION START
		if self.set[item]:
			del self.set[item]
		# SOLUTION END
